MSAL/src/public/MSALPublicClientApplication.h

```diff
@@ -34,50 +34,156 @@
 
 @interface MSALPublicClientApplication : NSObject
 
+/*! Whether or not to validate the authority used with the application, default YES */
 @property BOOL validateAuthority;
+
+/*! The authority used with the application object */
 @property (readonly) NSURL *authority;
+
+/*! The client ID of the application */
 @property (readonly) NSString *clientId;
+
+/*! The redirect URI of the application */
 @property (readonly) NSURL *redirectUri;
 
 /*! Used in logging callbacks to identify what component in the application
     called MSAL. */
 @property NSString *component;
 
+/*!
+    Initialize a MSALPublicClientApplication with a given clientID for the default
+    authority
+ 
+    @param  clientId    The clientID of your application, you should get this from the app portal.
+    @param  error       The error that occurred creating the application object, if any (optional)
+ */
 - (id)initWithClientId:(NSString *)clientId
                  error:(NSError * __autoreleasing *)error;
+
+/*!
+    Initialize a MSALPublicClientApplication with a given clientID and authority
+ 
+    @param  clientId    The clientID of your application, you should get this from the app portal.
+    @param  authority   The authority to use for the application
+    @param  error       The error that occurred creating the application object, if any, if you're
+                        not interested in the specific error pass in nil.
+ */
 - (id)initWithClientId:(NSString *)clientId
              authority:(NSString *)authority
                  error:(NSError * __autoreleasing *)error;
 
+/*!
+    Returna an array of users visible to this application
+ 
+    @param  error   The error that occured trying to retrieve users, if any, if you're
+                    not interested in the specific error pass in nil.
+ */
 - (NSArray <MSALUser *> *)users:(NSError * __autoreleasing *)error;
+
+/*!
+    Returns a specific user for the identifier given (received from a user object returned
+    in a previous acquireToken call)
+ 
+    @param  error   The error that occured trying to the user, if any, if you're
+                    not interested in the specific error pass in nil.
+ */
 - (MSALUser *)userForIdentifier:(NSString *)identifier
                           error:(NSError * __autoreleasing *)error;
 
 #pragma SafariViewController Support
 
+/*!
+    Returns whether or not the URL is a response to a MSAL web authentication
+    session. Call this before passing the URL into -handleMSALResponse:
+ */
 + (BOOL)isMSALResponse:(NSURL *)response;
+
+/*!
+    Pass a URL response from your application delegate's openURL handler into
+    MSAL for web authnetication sessions.
+ */
 + (void)handleMSALResponse:(NSURL *)response;
+
+/*!
+    Cancel a currently running interactive web authentication session.
+ */
 + (void)cancelCurrentWebAuthSession;
```

Comment:
need a bit more detail here. what should happen to the WebView when you call this? What about a passed-in WebView?